crates/shared/src/order_validation.rs

```diff
@@ -500,6 +505,15 @@ impl OrderValidating for OrderValidator {
         Ok(())
     }
 
+    /// Validates the provided app data, returning an [`OrderAppData`] if valid.
+    ///
+    /// The validation entails verifying that the app data is well formed and
+    /// it's size.
+    ///
+    /// * In the case the app data contains both an hash and the data, the data
```

Comment:
Why "an hash" everywhere?
```suggestion
    /// * In the case the app data contains both a hash and the data, the data
```